src/overlays/actors/ovl_Demo_Effect/z_demo_effect.c

```diff
@@ -20,121 +104,2126 @@ const ActorInit Demo_Effect_InitVars = {
+        interpolated = 1.0f;
+    }
+    return interpolated;
+}
+
+/**
+ * Initializes information for Jewel/Spritual Stone actors.
+ */
+void DemoEffect_InitJewel(GlobalContext* globalCtx, DemoEffect* this) {
+    this->initDrawFunc = &DemoEffect_DrawJewel;
+    if (LINK_IS_CHILD) {
+        this->initUpdateFunc = &DemoEffect_UpdateJewelChild;
+    } else {
+        this->initUpdateFunc = &DemoEffect_UpdateJewelAdult;
+    }
+    if (globalCtx->sceneNum == SCENE_TOKINOMA) {
+        Actor_SetScale(&this->actor, 0.35f);
+    } else {
+        Actor_SetScale(&this->actor, 0.10f);
+    }
+    this->csActionId = 1;
+    this->actor.shape.rot.x = 16384;
+    DemoEffect_InitJewelColor(this);
+    this->jewelCsRotation.z = 0;
+    this->jewel.opacity = 0;
+    this->jewelCsRotation.x = this->jewelCsRotation.y = this->jewelCsRotation.z;
+    sfxJewelId[0] = 0;
+}
+
+/**
+ * Initializes information for Get Item actors.
+ * These are the Medal and Light Arrow actors.
+ */
+void DemoEffect_InitGetItem(DemoEffect* this) {
+    this->getItem.isPositionInit = 0;
+    this->getItem.isLoaded = 0;
+    this->initDrawFunc = &DemoEffect_DrawGetItem;
+    this->initUpdateFunc = &DemoEffect_UpdateGetItem;
+    Actor_SetScale(&this->actor, 0.25f);
+    this->csActionId = 6;
+}
+
+/**
+ * Main Actor Init function
+ */
+void DemoEffect_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    DemoEffect* this = THIS;
+    s32 effectType;
+    s32 lightEffect;
+    s32 objectIndex;
+    DemoEffect* crystalLight;
+    DemoEffect* lightRing;
+
+    effectType = GET_EFFECT_TYPE(&this->actor);
+    lightEffect = GET_LIGHT_EFFECT_COLOR_PARAM(&this->actor);
+
+    osSyncPrintf("\x1b[36m no = %d\n\x1b[m", effectType);
+
+    objectIndex =
+        effectTypeObjects[effectType] == 1 ? 0 : Object_GetIndex(&globalCtx->objectCtx, effectTypeObjects[effectType]);
+
+    osSyncPrintf("\x1b[36m bank_ID = %d\n\x1b[m", objectIndex);
+
+    if (objectIndex < 0) {
+        __assert("0", "../z_demo_effect.c", 723);
+    } else {
+        this->initObjectBankIndex = objectIndex;
+    }
+
+    this->effectFlags = 0;
+    Actor_SetScale(&this->actor, 0.2f);
+
+    switch (effectType) {
+        case Demo_Effect_Crystal_Light:
+            this->initDrawFunc = &DemoEffect_DrawCrystalLight;
+            this->initUpdateFunc = &DemoEffect_UpdateCrystalLight;
+            break;
+
+        case Demo_Effect_Fire_Ball:
+            this->initDrawFunc = &DemoEffect_DrawFireBall;
+            this->initUpdateFunc = &DemoEffect_UpdatePositionToParent;
+            Actor_SetScale(&this->actor, 0.1f);
+            break;
+
+        case Demo_Effect_Blue_Orb:
+            this->initDrawFunc = &DemoEffect_DrawBlueOrb;
+            this->initUpdateFunc = &DemoEffect_UpdateBlueOrbGrow;
+            this->blueOrb.opacity = 255;
+            this->blueOrb.scale = 5;
+            this->blueOrb.rotation = 0;
+            Actor_SetScale(&this->actor, 0.05f);
+            this->primXluColor[0] = 0xBC;
```

Comment:
Are you sure these aren't `RGB8`s?

Reply:
I think that they are, but Fig and Roman told me the preference would be to make them arrays. They may not have known the full context, so I guess we'll let them chime in here now that the whole file is ready.

Reply:
iirc in order to use rgb8 you had to do some weird union? in which case the array is def preferred

Reply:
I wouldn't need a union in this case I believe.

Reply:
I would say to use `Color_RGB8` if you can. Array is only preferred over a union of both.